### PR TITLE
Epmctdm 7086 time improvement for clustering

### PR DESCRIPTION
**Pull request by Ijka**
sihlouette_score is time consuming method for optimal clusters number calculation.
Changed to Davies-Bouldin Index.

### CODE REVIEW

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 1]
```diff
@@ -965,21 +975,22 @@ def diversity(statistics):
         else:
             return 0
 
-    def __automated_silhouette(self):
-        silhouette_scores = []
+    @timing
+    def __automated_davies_bouldin(self):
```
Comment (rioman) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:979]: May be it makes sense to rename a function (and a scores in the next line) irrelevant to used algorithm?

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 2]
```diff
@@ -897,25 +898,44 @@ def calculate_all(self, categ_columns: List[str], cont_columns: List[str]):
...
         row_limit = min(len(self.original), len(self.synthetic))
+
         self.merged = (
             pd.concat(
                 [
-                    self.original[cont_columns + categ_columns].sample(row_limit),
-                    self.synthetic[cont_columns + categ_columns].sample(row_limit),
+                    self.original[cont_columns + categ_columns].sample(
+                        row_limit,
+                        random_state=10
```
Comment (rioman) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:919]: Do we still need random state here?

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 3]
```diff
@@ -897,25 +898,44 @@ def calculate_all(self, categ_columns: List[str], cont_columns: List[str]):
...
+                    ),
+                    self.synthetic[cont_columns + categ_columns].sample(
+                        row_limit,
+                        random_state=10
```
Comment (rioman) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:923]: And here

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 4]
```diff
@@ -897,25 +898,44 @@ def calculate_all(self, categ_columns: List[str], cont_columns: List[str]):
...
         )
+
         if len(self.merged) == 0:
             logger.warning(
                 "No clustering metric will be formed due to empty DataFrame"
             )
             return None
-        self.__preprocess_data()
-        optimal_clust_num = self.__automated_silhouette()
+        self.merged_transformed = self.__preprocess_data(self.merged)
```
Comment (rioman) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:937]: Why have you refactored the function to take an argument?

File: src/syngen/ml/metrics/metrics_classes/metrics.py (hunk 2 above)
Comment (rioman) [reply src/syngen/ml/metrics/metrics_classes/metrics.py:941]: Add pls #TODO reconsider necessity of random state

**Review by rioman — APPROVED** [review]

File: src/syngen/ml/metrics/metrics_classes/metrics.py (hunk 2 above)
Comment (Ijka) [reply src/syngen/ml/metrics/metrics_classes/metrics.py:942]: Done

**Pull request merged**